lib/scoring.rb

```diff
@@ -0,0 +1,82 @@
+#require_relative '../scrabble'  #   allows this class to access scrabble and run through irb
+
+class Scrabble::Scoring
+
+  attr_reader :word
+
+  include Scrabble
+
+  LETTER_POINTS = {
+    1 => ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+    2 => ["D", "G"],
+    3 => ["B", "C", "M", "P"],
+    4 => ["F", "H", "V", "W", "Y"],
+    5 => ["K"],
+    8 => ["J", "X"],
+    10 => ["Q", "Z"]
+  }
+
+  def initializer (word)
+    @word = word["word"]
+  end
+
+  # returns the score value for a single given word
+  def self.score(word)
+    score = 0
+    tiles = 0
+
+    word_array = word.upcase.split("")
+
+    word_array.each do |letter|
+      LETTER_POINTS.each do |points, letters_array|
+        if letters_array.include? letter
+          score += points
+          tiles += 1
+        end
+      end
+    end
+    # if all seven tiles are used the below tenary will add 50 points to the users score else return score
+    tiles >= 7 ? score += 50 : score
+    score
+  end
+
+  # returns the score value for a single given letter
+  def self.char_score(char)
+    score = 0
+
+    LETTER_POINTS.each do |points, letters_array|
+      if letters_array.include? char.upcase
+        score += points
+      end
+    end
+    score
+  end
+
+  def self.highest_score_from(array_of_words) # array_of_words = ["cats", "fee", "no"]
+
+    array_of_word_values = []
+    array_of_words.each do |word|
+      array_of_word_values << [word, self.score(word), word.length] #  = [["cats", 7, 4], ["fee", 7, 3], ["no", 2, 2]]
```

Comment:
Instead of using an array of arrays:
```
[["cats", 7, 4], ["fee", 7, 3], ["no", 2, 2]]
```

Consider using an array of hashes so you have context of each value
```
[
  {word: "cats", score: 7, length: 4}, 
  {word: "fee", score: 7, length: 3}, 
  {word: "no", score: 2, length: 2}
]
```